--- sundials-2.6.2/CMakeLists.txt	2015-07-31 02:07:21.000000000 +0200
+++ linux_x86_64/CMakeLists.txt	2016-05-04 16:59:46.082056444 +0200
@@ -42,6 +42,12 @@
 SET(PACKAGE_TARNAME "sundials")
 SET(PACKAGE_VERSION "2.6.2")
 
+if(UNIX AND ${CMAKE_SIZEOF_VOID_P} EQUAL 8)
+  # required for static builds on 64bit
+  add_compile_options(-fPIC)
+endif()
+
+
 # 
 SET_PROPERTY(GLOBAL PROPERTY USE_FOLDERS ON)
 
